### PR TITLE
Remove ConnectionManagerSettings from NodeSettings

### PR DESCRIPTION
**Pull request by quantumagi**
This PR removes the ConnectionManagerSettings reference from NodeSettings and re-introduces it via the BaseFeature.

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs [hunk 1]
```diff
@@ -152,7 +153,9 @@ public class TestContext
 
                 var peerAddressManager = new PeerAddressManager(nodeSettings.DataFolder, loggerFactory);
                 var peerDiscovery = new PeerDiscovery(new AsyncLoopFactory(loggerFactory), loggerFactory, Network.Main, networkPeerFactory, new NodeLifetime(), nodeSettings, peerAddressManager);
-                var connectionManager = new ConnectionManager(dateTimeProvider, loggerFactory, this.network, networkPeerFactory, nodeSettings, new NodeLifetime(), new NetworkPeerConnectionParameters(), peerAddressManager, new IPeerConnector[] { }, peerDiscovery);
+                var connectionSettings = new ConnectionManagerSettings();
+                connectionSettings.Load(nodeSettings);
+                var connectionManager = new ConnectionManager(dateTimeProvider, loggerFactory, this.network, networkPeerFactory, nodeSettings, new NodeLifetime(), new NetworkPeerConnectionParameters(), peerAddressManager, new IPeerConnector[] { }, peerDiscovery, connectionSettings);
```
Comment (dangershony) [inline src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs]: If we pass `nodeSettings` to ConnectionManager then we probably dont need to pass `connectionSettings`  as well right?

File: src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs [hunk 2]
```diff
@@ -49,12 +50,10 @@ public void PeerConnectorAddNode_FindPeerToConnectTo_Returns_AddNodePeers()
             peerAddressManager.AddPeer(networkAddressAddNode, IPAddress.Loopback);
             peerAddressManager.AddPeer(networkAddressDiscoverNode, IPAddress.Loopback);
 
-            var nodeSettings = new NodeSettings
-            {
-                ConnectionManager = new Configuration.Settings.ConnectionManagerSettings()
-            };
+            var nodeSettings = new NodeSettings();
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs]: Dont we need to instantiate connectionSettings? (using the Load method?)

File: src/Stratis.Bitcoin/Base/BaseFeature.cs [hunk 3]
```diff
@@ -302,6 +303,7 @@ public static IFullNodeBuilder UseBaseFeature(this IFullNodeBuilder fullNodeBuil
                     services.AddSingleton<INetworkPeerFactory, NetworkPeerFactory>();
                     services.AddSingleton<NetworkPeerConnectionParameters>(new NetworkPeerConnectionParameters());
                     services.AddSingleton<IConnectionManager, ConnectionManager>();
+                    services.AddSingleton<ConnectionManagerSettings>(new ConnectionManagerSettings());
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Base/BaseFeature.cs]: No need to create the instance here
Just do 
```
  services.AddSingleton<ConnectionManagerSettings>()
```

File: src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs (hunk 1 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs]: Ah I realise NodeSettings does not have  ConnectionManagerSettings anymore

**Comment by dangershony** [comment]
Some minor comments @fassadlr can you also review as this is a bit related to your changes.

File: src/Stratis.Bitcoin/Base/BaseFeature.cs (hunk 3 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin/Base/BaseFeature.cs]: +1

Comment (quantumagi) [reply src/Stratis.Bitcoin/Base/BaseFeature.cs]: Done.

File: src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs (hunk 2 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs]: Done.

File: src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs]: Ok.

**Comment by quantumagi** [comment]
@fassaldr, there are other features such as MemPool that names the settings as " mempoolSettings" on the MempoolManager so for the sake of consistency it may be preferable to also keep the naming "connectionSettings".

File: src/Stratis.Bitcoin.Features.Dns.Tests/GivenAWhitelistManager.cs [hunk 4]
```diff
@@ -78,20 +79,22 @@ public void WhenConstructorCalled_AndDnsServerIsNull_ThenArgumentNullExceptionIs
...
 
             // Act and Assert.
-            a.ShouldThrow<ArgumentNullException>().Which.Message.Should().Contain("nodeSettings");
+            a.ShouldThrow<ArgumentNullException>().Which.Message.Should().Contain("connectionSettings");
         }
-
+/*
         [Fact]
         [Trait("DNS", "UnitTest")]
         public void WhenConstructorCalled_AndNodeSettingsDnsHostNameIsNull_ThenArgumentNullExceptionIsThrown()
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Dns.Tests/GivenAWhitelistManager.cs]: If this test is never going to be applicable again, we can just remove it altogether 👍

File: src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs [hunk 5]
```diff
@@ -63,26 +69,22 @@ public class WhitelistManager : IWhitelistManager
...
 
             this.dateTimeProvider = dateTimeProvider;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.peerAddressManager = peerAddressManager;
             this.dnsServer = dnsServer;
-            this.dnsPeerBlacklistThresholdInSeconds = dnsSettings.DnsPeerBlacklistThresholdInSeconds;
-            this.dnsHostName = dnsSettings.DnsHostName;
-            this.externalEndpoint = nodeSettings.ConnectionManager.ExternalEndpoint;
-            this.fullNodeMode = dnsSettings.DnsFullNode;
+            this.dnsSettings = dnsSettings;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs]: Yes, this is what I was getting at in the previous PRs! Good job.

File: src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs [hunk 6]
```diff
@@ -92,6 +94,10 @@ public void RefreshWhitelist()
         {
             this.logger.LogTrace("()");
 
+            this.dnsPeerBlacklistThresholdInSeconds = this.dnsSettings.DnsPeerBlacklistThresholdInSeconds;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs]: Now we initialize the WhiteListManager's settings in the proper place.

File: src/Stratis.Bitcoin.Features.Dns.Tests/GivenAWhitelistManager.cs (hunk 4 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Dns.Tests/GivenAWhitelistManager.cs]: Done.

File: src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs (hunk 5 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs]: Yes, I saw the light. Thank you.

File: src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs (hunk 6 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Dns/WhitelistManager.cs]: Yes.

**Pull request merged**